src/utils/src/Math.c

```diff
@@ -0,0 +1,30 @@
+#include "Include_i.h"
+
+/**
+ * Calculates power using Divide and Conquer style - O(log exponent) complexity
+ *
+ * WARNING:
+ * The result could overflow if the base ^ power exceeds UINT64 limit.
+ * It is the callers responsibility to sanitize the values before calling this API
+ */
+UINT64 power(UINT32 base, UINT32 exponent) {
```

Comment:
My only concern on this is the name collision. Consider using a longer name perhaps.  Also, think if we want to change the API signature to return STATUS to indicate bad values passed in. For example, base being less than 2 and greater that some large value. Handling the overflow would also need to return a status code - something like invalid args or similar.

Reply:
I can change the API name. 
I though about adding validations for input but I feel it is very ambiguous. We can have larger base smaller exponent and smaller base and a relatively larger exponent. It's pretty hard to decide what's right OR wrong. That's why I've added a warning comment and left it to the developer's discretion.

Reply:
+1 about changing the API name, I also think we need some way to enforce we will not overflow

Reply:
Added checks to prevent overflows. And changed API name too.